### ISSUE TITLE
[Quest bug] Blightbeats be Damned!

### ISSUE THREAD

**Issue by StormBytePP** [TrinityCore/TrinityCore#15266]
In this quest,http://www.wowhead.com/quest=12072/blightbeasts-be-damned, you get a flare that summons a mount to be able to fly and attack requested mobs.

The problem, is that you can't do the quest as you get randomly unmounted (not just when you are over water).

To reproduce, just be lvl 71+ take the quest, or do .quest add 12072, be in the correct zone, and use the flare.

Tested in rev 9b285d260e1cad66ce6e2b05948c5b4fb2850466 (3.3.5) but it may also apply to 6.x)

**Comment by Rushor** [comment]
maybe the vehicle itself is causing this issue

**Comment by Killyana** [comment]
Yes it comes from the vehicle and it can maybe fixed by hacking the spell attributes.

**Comment by StormBytePP** [comment]
@Killyana With "hardcoded riding spell" can you use also vehicle's abilities?

**Comment by Killyana** [comment]
Yes

**Comment by Rushor** [comment]
differneces from both spells:
![Bild](http://i.imgur.com/3fZ4biC.png)

**Comment by Killyana** [comment]
Looks like the problem comes from the Interrupt Flags.

**Comment by Rushor** [comment]
AURA_INTERRUPT_FLAG_NOT_ABOVEWATER      = 0x00000080,   // 7    removed by entering water
:/

**Comment by Rushor** [comment]
```diff
diff --git a/src/server/game/Spells/SpellMgr.cpp b/src/server/game/Spells/SpellMgr.cpp
index 4dde21f..f7b301a 100644
--- a/src/server/game/Spells/SpellMgr.cpp
+++ b/src/server/game/Spells/SpellMgr.cpp
@@ -3697,6 +3697,9 @@ void SpellMgr::LoadSpellInfoCorrections()
             case 60068: // Hunters mark
                 spellInfo->AuraInterruptFlags = AURA_INTERRUPT_FLAG_TAKE_DAMAGE;
                 break;
+            case 47424: // Korkron Wing Commander
+                spellInfo->AuraInterruptFlags = 0;
+                break;
             case 62225: // Risen Ally - effect doesn't work correctly
                 spellInfo->Effects[EFFECT_1].Effect = 0;
                 break;
```

http://i.imgur.com/H1IfkXL.jpg

**Comment by Killyana** [comment]
In retail the spell cannot be casted if the player is in the water, but once mounted the aura will not be removed if he fly over the water.
Some comments:
- Don't stand in water it will be bugged and you will have to wait 3 min.
- Make sure if your a Deathknight or a Shaman don't use water walk then go out on the water and use flare.. the wyrn will not pick you up he just dives into the water and you get to wait 2 and half mins.

**Comment by Rushor** [comment]
the it seems that it worked on beta (wtbvideo) - it get bugged in 3.3.5 during 2010 - and they refixed it in >= 4.x expansions?

https://www.youtube.com/watch?feature=player_detailpage&v=KoSP-CRj-J0#t=105
https://www.youtube.com/watch?feature=player_detailpage&v=vEdnCOptCcE#t=235

blizz is fucked up xD

**Comment by StormBytePP** [comment]
I just hope that to be blizzlike != to have same bugs :)

**Comment by StormBytePP** [comment]
@Rushor @Killyana I found out that:
```diff
diff --git a/src/server/game/Entities/Unit/Unit.cpp b/src/server/game/Entities/Unit/Unit.cpp
index 8656bd0..61dfe08 100644
--- a/src/server/game/Entities/Unit/Unit.cpp
+++ b/src/server/game/Entities/Unit/Unit.cpp
@@ -3194,6 +3194,9 @@ void Unit::UpdateUnderwaterState(Map* m, float x, float y, float z)
 {
     if (!IsPet() && !IsVehicle())
         return;
+    
+    if (IsFlying())
+        return;
 
     LiquidData liquid_status;
     ZLiquidStatus res = m->getLiquidStatus(x, y, z, MAP_ALL_LIQUIDS, &liquid_status);
```
Fixes the problem of dismounting when above water. But not sure which side effects it may have. Any hints?

**Comment by StormBytePP** [comment]
Fixed by rev 70f029f
